Build broken on main: cfg_select! requires Rust 1.95, toolchain pins 1.92.0

# build: rusqlite 0.40.0 bump pulls in `cfg_select!` (Rust 1.95) but the toolchain is pinned to Rust 1.92.0

## Summary

`cargo build` / `nix build` fails on the latest `main` (`17a8d37`).

The transitive dependency `libsqlite3-sys 0.38.0` uses the `cfg_select!` macro in its `build.rs`, but that macro was **stabilized in Rust 1.95**. This repository pins **`channel = "1.92.0"`** in `rust-toolchain.toml`, so `cfg_select!` is treated as an unstable feature and the build fails to compile.

## Offending commit

- Last working commit: `22598d35` (2026-05-24) — rusqlite 0.39.0 line
- Breaking commit: **`3b0f8f35` (2026-05-31) `chore(deps): bump rusqlite from 0.39.0 to 0.40.0 (#139)`**
  - rusqlite 0.40.0 pulls in `libsqlite3-sys 0.38.0`
  - `libsqlite3-sys 0.38.0`'s `build.rs` uses `cfg_select!`
  - `cfg_select!` was stabilized in Rust 1.95 ([PR #149783](https://github.com/rust-lang/rust/pull/149783))

## Error log

```text
error[E0658]: use of unstable library feature `cfg_select`
   --> .../libsqlite3-sys-0.38.0/build.rs:110:9
    |
110 |         cfg_select! {
    |         ^^^^^^^^^^
    |
    = note: see issue #115585 <https://github.com/rust-lang/rust/issues/115585> for more information

For more information about this error, try `rustc --explain E0658`.
error: could not compile `libsqlite3-sys` (build script) due to 1 previous error
```

## Steps to reproduce

```sh
cargo build
```

## Environment

- platform: aarch64-darwin (macOS)
- pinned Rust: `rust-toolchain.toml` → `channel = "1.92.0"`

## Suggested fixes (either one)

1. Bump `channel` in `rust-toolchain.toml` to `1.95.0` or later (to keep up with the newer dependency that uses `cfg_select!`)
2. Revert `rusqlite` back to `0.39.0` (revert #139 to avoid `libsqlite3-sys 0.38.0`)

--- a/rust-toolchain.toml
+++ b/rust-toolchain.toml
@@ -1,2 +1,2 @@
 [toolchain]
-channel = "1.92.0"
+channel = "1.95.0"

--- a/src/version/registries/crates_io.rs
+++ b/src/version/registries/crates_io.rs
@@ -96,7 +96,7 @@ impl Registry for CratesIoRegistry {
             })
             .collect();
 
-        versions.sort_by(|(_, a), (_, b)| a.cmp(b));
+        versions.sort_by_key(|(_, a)| *a);
 
         let versions: Vec<String> = versions.into_iter().map(|(v, _)| v).collect();
 

--- a/src/version/registries/github.rs
+++ b/src/version/registries/github.rs
@@ -132,7 +132,7 @@ impl Registry for GitHubRegistry {
             })
             .collect();
 
-        releases_with_dates.sort_by(|(_, a), (_, b)| a.cmp(b));
+        releases_with_dates.sort_by_key(|(_, a)| *a);
 
         let versions = releases_with_dates
             .into_iter()

--- a/src/version/registries/jsr.rs
+++ b/src/version/registries/jsr.rs
@@ -109,7 +109,7 @@ impl Registry for JsrRegistry {
             })
             .collect();
 
-        versions.sort_by(|(_, a), (_, b)| a.cmp(b));
+        versions.sort_by_key(|(_, a)| *a);
 
         let versions: Vec<String> = versions.into_iter().map(|(v, _)| v).collect();
 

--- a/src/version/registries/npm.rs
+++ b/src/version/registries/npm.rs
@@ -109,7 +109,7 @@ impl Registry for NpmRegistry {
             })
             .collect();
 
-        versions.sort_by(|(_, a), (_, b)| a.cmp(b));
+        versions.sort_by_key(|(_, a)| *a);
 
         let versions: Vec<String> = versions.into_iter().map(|(v, _)| v).collect();
 

